From 1645756137221c6cdb45caffb811681a7c47732c Mon Sep 17 00:00:00 2001
From: Tsukasa OI <research_trasio@irq.a4lg.com>
Date: Sat, 12 Aug 2023 00:38:18 +0000
Subject: [PATCH 01/30] RISC-V: Add stub support for existing extensions
 (privileged)

gcc/ChangeLog:

	* common/config/riscv/riscv-common.cc
	(riscv_implied_info): Add implications from privileged
extensions.
	(riscv_ext_version_table): Add stub support for all privileged
	extensions supported by Binutils.
---
 gcc/common/config/riscv/riscv-common.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/gcc/common/config/riscv/riscv-common.cc b/gcc/common/config/riscv/riscv-common.cc
index e19c0985ca0..68c2e843b4e 100644
--- a/gcc/common/config/riscv/riscv-common.cc
+++ b/gcc/common/config/riscv/riscv-common.cc
@@ -107,7 +107,7 @@ static const riscv_implied_info_t riscv_implied_info[] =
   
   {"zhinx", "zhinxmin"},
   {"zhinxmin", "zfinx"},
-
+  {"sscofpmf", "zicsr"},
   {NULL, NULL}
 };
 
@@ -219,8 +219,10 @@ static const struct riscv_ext_version riscv_ext_version_table[] =
 
   {"zmmul", ISA_SPEC_CLASS_NONE, 1, 0},
 
+  {"sscofpmf",  ISA_SPEC_CLASS_NONE, 1, 0},
   {"svinval", ISA_SPEC_CLASS_NONE, 1, 0},
   {"svnapot", ISA_SPEC_CLASS_NONE, 1, 0},
+  {"svpbmt",  ISA_SPEC_CLASS_NONE, 1, 0},
 
   {"xtheadba", ISA_SPEC_CLASS_NONE, 1, 0},
   {"xtheadbb", ISA_SPEC_CLASS_NONE, 1, 0},
-- 
2.25.1

